### PR TITLE
Add an example and instructions / script for running it

### PR DESCRIPTION
**Pull request by makaimann**
This PR adds an example usage of the API with a script and examples for how to run it.

This could have been incorporated into the CMake infrastructure, but I thought it was nice to show how someone might use it outside of CMake for their own project. It gives a better understanding of what happens under the hood.

### CODE REVIEW

File: examples/Makefile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+all: cvc4_qf_ufbv btor_qf_ufbv
+
+# Note: assumes smt-switch has been installed in a directory called
+# example-install in this directory
```
Comment (yoni206) [inline examples/Makefile:4]: maybe mention here that the script `build.sh` does just that?

File: examples/Makefile [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+cvc4_qf_ufbv: cvc4_qf_ufbv.cpp
```
Comment (yoni206) [inline examples/Makefile:6]: CVC4 will take time to install. This is unrelated to this PR, btu maybe in the artifact you can have it already installed in deps.

File: examples/README.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Examples
+
+This directory contains example usage of smt-switch.
+
+## QF_UFBV
+`QF_UFBV` stands for quantifier-free uninterpreted functions and bit-vectors. The files `cvc4_qf_ufbv.cpp` and `btor_qf_ufbv.cpp` demonstrate some common usage of the API with this logic. The `build.sh` script will setup and install smt-switch in this directory and then build the examples. The script has comments explaining each step for building smt-switch. This assumes that you have all the necessary dependencies other than the underlying solvers and smt-switch already installed.  You can look at the `Makefile` to understand how to link smt-switch to a binary. If the script succeeds in building everything, you can run each of the files with `./cvc4_qf_ufbv.out` and `./btor_qf_ufbv.out`, respectively. Note that there may be a delay for looking up the shared libraries on some systems the first time they are run. Importantly, notice that the files differ by only two lines, demonstrating that changing the solver is straightforward.
```
Comment (yoni206) [inline examples/README.md:6]: Can you maybe split this to several lines in a meaningful way?

Comment (yoni206) [inline examples/README.md:6]: Maybe write a warning that running the build script will take time, depending on the environment, and give an estimate?

Comment (yoni206) [inline examples/README.md:6]: I actually used the script and it failed because i had an old version of CVC4 lying around in deps. To prevent that, maybe install the solvers inside the newly created examples dir?

File: examples/README.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# Examples
+
+This directory contains example usage of smt-switch.
+
+## QF_UFBV
```
Comment (yoni206) [inline examples/README.md:5]: can u explain here how one can change the cpp recompile and rerun?

**Review by yoni206 — COMMENTED** [review]
Thanks for doing this, i think it will be very useful for users.

I have some comments inline, and also another one -- maybe link to this file from the main README?

File: examples/Makefile (hunk 1 above)
Comment (makaimann) [reply examples/Makefile]: Good idea -- addressed with c2b862a13b2101020b66ded1942118f9002f718f

File: examples/Makefile (hunk 2 above)
Comment (makaimann) [reply examples/Makefile:6]: good point, I'll make sure it's installed.

File: examples/README.md (hunk 3 above)
Comment (makaimann) [reply examples/README.md]: Good idea, done with c2b862a13b2101020b66ded1942118f9002f718f

Comment (makaimann) [reply examples/README.md]: Since we decided offline to separate building the backend solvers this shouldn't be a problem anymore.

Comment (makaimann) [reply examples/README.md]: Ditto. Handled by removing that from the build script.

File: examples/README.md (hunk 4 above)
Comment (makaimann) [reply examples/README.md:5]: Great idea -- done with c2b862a13b2101020b66ded1942118f9002f718f.

**Review by yoni206 — APPROVED** [review]
